internal/controller/dnsrecord_controller.go

```diff
@@ -156,15 +156,25 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 }
 
 func (r *DNSRecordReconciler) updateStatus(ctx context.Context, previous, current *v1alpha1.DNSRecord, requeueAfter time.Duration) (reconcile.Result, error) {
-	current.Status.QueuedFor = metav1.NewTime(reconcileStart.Add(requeueAfter))
+	if requeueAfter != noRequeueDuration {
+		current.Status.QueuedFor = metav1.NewTime(reconcileStart.Add(requeueAfter))
+	}
 
 	if !equality.Semantic.DeepEqual(previous.Status, current.Status) {
-		updateError := r.Status().Update(ctx, current)
-		if apierrors.IsConflict(updateError) {
-			return ctrl.Result{Requeue: true}, nil
+		if updateError := r.Status().Update(ctx, current); updateError != nil {
+			if apierrors.IsConflict(updateError) {
+				return ctrl.Result{Requeue: true}, nil
+			}
+			return ctrl.Result{}, updateError
 		}
-		return ctrl.Result{RequeueAfter: requeueAfter}, updateError
 	}
+
+	if requeueAfter == noRequeueDuration {
+		// `ctrl.Result{RequeueAfter: 0, Requeue: true}, nil` results in the same backoff behaviour as `ctrl.Result{}, err`
+		// but with no error reported in the logs
+		return ctrl.Result{RequeueAfter: 0, Requeue: true}, nil
+	}
```

Comment:
So the `noRequeueDuration` isn't really a no requeue anymore? Should we call it an `errorRequeueDuration` instead? And the `return ctrl.Result{RequeueAfter: 0, Requeue: true}, nil` is confusing a bit. It will still requeue immediately as the boolean flag takes precedence but `RequeueAfter: 0` is:
1. Ignored if `Reqeue: true` 
2. Means "do not requeue" if `Requeue: false` 
I'd vote for `return ctrl.Result{Requeue: true}, nil` to be less confusing